projects/igniteui-angular/src/lib/grids/toolbar/grid-toolbar.component.ts

```diff
@@ -325,9 +343,24 @@ export class IgxGridToolbarComponent extends DisplayDensityBase implements After
      * ```
      */
     public exportToCsvClicked() {
+        let exportEnded = false;
+        this.csvExporter.onExportEnded.subscribe((args: any) => {
+            exportEnded = true;
+            this.setIsExporting(false);
+        });
+        setTimeout(() => {
+            if (!exportEnded) {
+                this.setIsExporting(true);
+            }
+        }, 500);
```

Comment:
We should move that logic inside `performExport` method in order to avoid duplication. We should also make sure to not display progress indicator if the exporting event was canceled.